### PR TITLE
fix(lock): fix unlocking of auto generated items

### PR DESCRIPTION
**Pull request by petrjasek**
when item is saved it gets populated, but before it's not set
and unlocking it fails with KeyError. there is also some bc fix
for items using objectId and not uuid

### CODE REVIEW

**Comment by ioanpocol** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:

File: server/apps/common/models/io/base_proxy.py [hunk 1]
```diff
@@ -49,11 +49,11 @@ def delete(self, resource, filter):
...
-        _id = doc.get(ID_FIELD, None)
-        if ID_FIELD in doc:
-            del doc[ID_FIELD]
+        _id = doc.pop(ID_FIELD, None)
         original = self.find_one(resource, filter, None)
-        res = getattr(self.data_layer, method)(resource, filter[ID_FIELD], doc, original)
-        if _id is not None:
-            doc[ID_FIELD] = _id
+        if filter[ID_FIELD] != original[ID_FIELD]:
+            filter[ID_FIELD] = original[ID_FIELD]  # make sure it's correct type
```
Comment (plamut) [inline server/apps/common/models/io/base_proxy.py]: The `if`condition seems kind of redundant here.

**Comment by plamut** [comment]
:+1:

File: server/apps/common/models/io/base_proxy.py (hunk 1 above)
Comment (petrjasek) [inline server/apps/common/models/io/base_proxy.py]: ah right

**Pull request merged**